src/coreclr/gc/gc.cpp

```diff
@@ -4964,6 +4964,25 @@ BOOL gc_heap::reserve_initial_memory (size_t normal_size, size_t large_size, siz
         }
     }
 
+    if (reserve_success && separated_poh_p)
+    {
+        for (int heap_no = 0; reserve_success && heap_no < num_heaps; heap_no++)
+        {
+            if (!GCToOSInterface::VirtualCommit(memory_details.initial_pinned_heap[heap_no].memory_base, pinned_size))
+            {
+                reserve_success = FALSE;
+            }
+        }
+        if (!reserve_success)
+        {
+            for (int heap_no = 0; reserve_success && heap_no < num_heaps; heap_no++)
```

Comment:
we can allocate memory in different fashion, eg, on each NUMA node (in which case you'll need to free by blocks on each node). we could also allocate the whole thing in one chunk.